burner-wallet-plugin/tokenbridge-bw-exchange/src/utils/utils.ts

```diff
@@ -34,11 +34,23 @@ export const waitForEvent = async (web3, contract: Contract, event: string, call
   }
 }
 
-export const isBridgeContract = async (contract: Contract): Promise<boolean> => {
+export const isVanillaBridgeContract = async (contract: Contract): Promise<boolean> => {
   try {
     await contract.methods.deployedAtBlock().call()
     return true
   } catch (e) {
     return false
   }
 }
+
+export const isBridgeContract = async (contract: Contract, allowedModes?: string[]): Promise<boolean> => {
+  try {
+    const mode = await contract.methods.getBridgeMode().call()
+    if (typeof allowedModes === 'undefined') {
```

Comment:
what is the use case for `undefined`?

Reply:
If `allowedModes` is not passed to this function, then the function will consider any value returned by `getBridgeMode()` as allowed one.

Reply:
ok. thanks